mrp_operations_extension/models/mrp_production.py

```diff
@@ -157,3 +157,14 @@ def action_start_working(self):
         if self.production_id.state in ('confirmed', 'ready'):
             self.production_id.state = 'in_production'
         return super(MrpProductionWorkcenterLine, self).action_start_working()
+
+    def action_done(self):
+        for move_line in self.move_lines:
```

Comment:
As exercise, you should start thinking with new API possibilities: 
`self.move_line.filtered(lambda x: x not in ('cancel', 'done')).action_cancel()`